upb/msg_internal.h

```diff
@@ -198,16 +198,16 @@ typedef struct {
   int ext_count;
 } upb_msglayout_file;
 
-// Computes a bitmask in which the |n| lowest bits are set, except that we
-// skip the lowest bit (because upb never uses hasbit 0).
+// Computes a bitmask in which the |l->required_count| lowest bits are set,
+// except that we skip the lowest bit (because upb never uses hasbit 0).
 //
 // Sample output:
 //    requiredmask(1) => 0b10 (0x2)
 //    requiredmask(5) => 0b111110 (0x3e)
 UPB_INLINE uint64_t upb_msglayout_requiredmask(const upb_msglayout *l) {
   int n = l->required_count;
-  assert(0 < n && n < 63);
-  return ((1 << n) - 1) << 1;
+  assert(0 < n && n <= 63);
+  return (((uint64_t)1 << n) - 1) << 1;
```

Comment:
oh god, now I have to remember the precedence of cast vs `<<`...  Can We get another set of parens?  The expression is getting a bit ridiculous I know....  Alternately you could break it into multiple statements:

```
uint64_t r = 1;
r <<= n;
r -= 1;
r << = 1;
return r;
```

I don't love it so feel free to go either way

Reply:
I changed it to `1ULL`.